app/controllers/saml_idp_controller.rb

```diff
@@ -147,6 +154,21 @@ def matching_cert_serial
     nil
   end
 
+  def sha256_alg_matching_cert
+    # if sha256_alg_matching_cert is nil, fallback to the "first" cert
+    saml_request.sha256_validation_matching_cert ||
+      saml_request_service_provider&.ssl_certs&.first
+  rescue SamlIdp::XMLSecurity::SignedDocument::ValidationError
+  end
+
+  def sha256_alg_matching_cert_serial
+    sha256_alg_matching_cert&.serial&.to_s
+  end
+
+  def certs_different?
+    encryption_cert != sha256_alg_matching_cert
```

Comment:
silly Q, are these cert objects that are OK with equality? I forget if OpenSSL certs do this, but sometimes certs contrsucted from the same string, that are different objects are still not equal

Reply:
i think they are ok with equality! i wrote a test that is passing, but just to double check i went into the console and saved the text of a certificate to two variables

(omitting a bunch of output)
```bash
> cert_text1 = Base64.encode64(saml_request.matching_cert.to_pem)
> cert_text2 = Base64.encode64(saml_request.matching_cert.to_pem)
> c1 = OpenSSL::X509::Certificate.new(Base64.decode64(cert_text1))
> c2 = OpenSSL::X509::Certificate.new(Base64.decode64(cert_text2))
> c1 == c2
=> true
```

Reply:
sweeeet thanks for confirming!